src/Sylius/Behat/Context/Api/Shop/CartContext.php

```diff
@@ -311,6 +312,26 @@ public function thisItemShouldHaveCode(array $item, string $variantCode): void
         );
     }
 
+    /**
+     * @Then /^(product "[^"]+") price should be decreased by ("[^"]+")$/
+     */
+    public function itsPriceShouldBeDecreasedBy(ProductInterface $product, int $amount): void
+    {
+        $this->compareItemSubtotal($product, function (ChannelPricingInterface $channelPricing) use ($amount) {
+            return $channelPricing->getPrice() - $amount;
```

Comment:
Won't it be easier if we would just pass an amount of discount as a second argument and make the calculation from this anonymous function in the private method?